corehq/apps/app_manager/suite_xml/post_process/endpoints.py

```diff
@@ -76,11 +77,28 @@ def _make_session_endpoint(self, endpoint_id, module, form=None, should_add_last
             if isinstance(child, CommandId):
                 frame.add_command(child.to_command())
             elif child.id in argument_ids:
-                self._add_datum_for_arg(frame, child.id)
+                self._add_datum_for_arg(frame, child)
+
+        def get_child(id, children):
```

Comment:
Nit: It's not a good idea to use `id` as a variable name since it shadows the builtin [`id`](https://docs.python.org/3/library/functions.html#id) function. 

[Shadowing](https://en.wikipedia.org/wiki/Variable_shadowing) in general is not good since it harder to read the code and can lead to bugs later if the code is changed e.g. if one of the usages of the variable is missed during a rename.

Same applies to the for loop on line 88.

Reply:
https://github.com/dimagi/commcare-hq/pull/33275/commits/16f6d1ce6ece0d52951a5a2be599b4deef422fbd and relatedly https://github.com/dimagi/commcare-hq/pull/33275/commits/68dbd184dfc10eaa30991ef26fe5e24a0d5b25f5